### PR TITLE
init port to FreeBSD

### PR DESCRIPTION
**Pull request by sistillson**
This is what you need to work on freebsd. still rough, not sure if fuse is working....

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,12 +2,27 @@
...
+CXX	:= c++ -std=c++11
+else
 CC	:= gcc -std=c99
 CXX	:= g++ -std=c++11
+endif
 CFLAGS	:= -Wall -O2 -flto -DVERSION=$(VERSION)
-LDFLAGS	:=
 
+ifeq ($(shell uname -s),FreeBSD)
+LDFLAGS	:= -L/usr/local/lib
```
Comment (terpstra) [inline Makefile]: Can you tell me which library was not found without this -L/usr/local/lib?

File: Makefile [hunk 2]
```diff
@@ -2,12 +2,27 @@
...
+else
+LDFLAGS	:= 
+endif
+
+ifeq ($(shell uname -s),FreeBSD)
+LOCAL_CFLAGS :=	-Iutf8proc -Igopt -Icommon -I/usr/local/include
```
Comment (terpstra) [inline Makefile]: Similarly, which headers were not found?

File: Makefile [hunk 3]
```diff
@@ -20,13 +35,22 @@ CORE_LDFLAGS :=	$(shell pkg-config --silence-errors --libs sqlite3 || echo -lsql
 		$(shell pkg-config --silence-errors --libs ncurses tinfo || pkg-config --silence-errors --libs ncurses || echo -lncurses)
 
 COMMON := common/jlexer.o $(patsubst %.cpp,%.o,$(wildcard common/*.cpp))
+ifeq ($(shell uname -s),FreeBSD)
+WAKE_ENV := WAKE_PATH=$(shell dirname $(shell which $(firstword $(CC)))):/usr/local/bin
```
Comment (terpstra) [inline Makefile]: ... and which programs/executables?

File: fuse/fuse.cpp [hunk 4]
```diff
@@ -73,7 +73,7 @@ int main(int argc, char *argv[])
 		pid_t pid = fork();
 		if (pid == 0) {
 			ofs.close();
-			execl(daemon.c_str(), "fuse-waked", mpath.c_str(), 0);
+			execl(daemon.c_str(), "fuse-waked", mpath.c_str(), (void *)0);
```
Comment (terpstra) [inline fuse/fuse.cpp]: What was the error that required (void *)?

**Comment by sistillson** [comment]
I think this version is more to show that it can easily be done and that it can be done without gcc. I wish it were straightforward to get rid of gmake (I always feel it's best to use as little gnu code as possible on a BSD system)

As to a system to use, I just use virtual box and download an image from https://www.freebsd.org/where.html (just one of the regular install dvd isos) and go from there.

I did the build on 12.0, but 11.3 should be pretty similar.

as root (or install sudo)
pkg install gmake pkgconf re2 sqlite fusefs-libs

If you aren't used to freebsd, you will probably be very frustrated. It's not so hand-holdy as linux but it's super consistent.

**Comment by terpstra** [comment]
I've merged an update for 12.0.

**Pull request closed without merging**